cloudinit/config/cc_raspberry_pi.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (C) 2024-2025, Raspberry Pi Ltd.
+#
+# Author: Paul Oberosler <paul.oberosler@raspberrypi.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import logging
+
+from cloudinit import subp
+from cloudinit.cloud import Cloud
+from cloudinit.config import Config
+from cloudinit.config.schema import MetaSchema
+from cloudinit.settings import PER_INSTANCE
+
+LOG = logging.getLogger(__name__)
+RPI_BASE_KEY = "rpi"
+RPI_INTERFACES_KEY = "interfaces"
+ENABLE_RPI_CONNECT_KEY = "enable_rpi_connect"
+SUPPORTED_INTERFACES = {
+    "spi": "do_spi",
+    "i2c": "do_i2c",
+    "serial": "do_serial",
+    "onewire": "do_onewire",
+    "remote_gpio": "do_rgpio",
+}
+RASPI_CONFIG_SERIAL_CONS_FN = "do_serial_cons"
+RASPI_CONFIG_SERIAL_HW_FN = "do_serial_hw"
+
+meta: MetaSchema = {
+    "id": "cc_raspberry_pi",
+    "distros": ["raspberry-pi-os"],
+    "frequency": PER_INSTANCE,
+    "activate_by_schema_keys": [RPI_BASE_KEY],
+}
+
+
+def configure_rpi_connect(enable: bool) -> None:
+    LOG.debug("Configuring rpi-connect: %s", enable)
+
+    num = 0 if enable else 1
+
+    try:
+        subp.subp(["/usr/bin/raspi-config", "do_rpi_connect", str(num)])
+    except subp.ProcessExecutionError as e:
+        LOG.error("Failed to configure rpi-connect: %s", e)
+
+
+# TODO: test
```

Comment:
At the very least some happy-path testing of the high level functions would be good here.

Reply:
we can remove this now since it's being worked on.